src/line_absorption.jl

```diff
@@ -44,58 +43,73 @@ function line_absorption!(α, linelist, λs::Wavelengths, temps, nₑ, n_densiti
         @error "Atomic hydrogen should not be in the linelist. Korg has built-in hydrogen lines."
     end
 
-    # preallocate some arrays for the core loop. 
-    # Each element of the arrays corresponds to an atmospheric layer, same at the "temps" array and 
-    # the values in "number_densities"
-    Γ = Vector{eltype(α)}(undef, size(temps))
-    γ = Vector{eltype(α)}(undef, size(temps))
-    σ = Vector{eltype(α)}(undef, size(temps))
-    amplitude = Vector{eltype(α)}(undef, size(temps))
-    levels_factor = Vector{eltype(α)}(undef, size(temps))
-    ρ_crit = Vector{eltype(α)}(undef, size(temps))
-    inverse_densities = Vector{eltype(α)}(undef, size(temps))
-    @showprogress desc="calculating line opacities" enabled=verbose for line in linelist
-        m = get_mass(line.species)
-
-        # doppler-broadening width, σ (NOT √[2]σ)
-        σ .= doppler_width.(line.wl, temps, m, ξ)
-
-        # sum up the damping parameters.  These are FWHM (γ is usually the Lorentz HWHM) values in 
-        # angular, not cyclical frequency (ω, not ν).
-        Γ .= line.gamma_rad
-        if !ismolecule(line.species)
-            @. Γ += nₑ * scaled_stark.(line.gamma_stark, temps)
-            Γ .+= n_densities[species"H_I"] .* scaled_vdW.(Ref(line.vdW), m, temps)
+    chunk_size = max(1, length(linelist) ÷ (tasks_per_thread * Threads.nthreads()))
```

Comment:
Let `n_chunks` denote `tasks_per_thread * Threads.nthreads()`.

As currently written, the code only splits `linelist` into `n_chunks` iff `length(linelist)` is an integer multiple of `n_chunks`. In the more common case where `length(linelist) % n_chunks > 0`, `linelist` will be split into `n_chunks+1`. In the limit where each chunk is the same amount of work, one of of your threads will have an extra task.

The following tweak ensures that you always split `linelist` into `n_chunks`, when `length(linelist) % n_chunks > 0`. (As you're probably aware, this is modification is cheap since most compilers know how to group nearby divisions and remainder operations into a single instruction).

```suggestion
    n_chunks = tasks_per_thread * Threads.nthreads()
    chunk_size = max(1, length(linelist) ÷ n_chunks + (length(linelist) % n_chunks > 0))
```

Reply:
Great catch, thanks!